rcl/src/rcl/node.c

```diff
@@ -169,15 +166,9 @@ rcl_node_init(
   }
   if (validation_result != RMW_NAMESPACE_VALID) {
     const char * msg = rmw_namespace_validation_result_string(validation_result);
-    if (!msg) {
-      char fixed_msg[256];
-      rcutils_snprintf(
-        fixed_msg, sizeof(fixed_msg),
-        "unknown validation_result '%d', this should not happen", validation_result);
-      RCL_SET_ERROR_MSG(fixed_msg, *allocator);
-    } else {
-      RCL_SET_ERROR_MSG(msg, *allocator);
-    }
+    char fixed_msg[256];
+    rcutils_snprintf(fixed_msg, sizeof(fixed_msg), "%s, result '%d'", msg, validation_result);
```

Comment:
I think these 2 lines are not necessary as we get a valid null-terminated string from `rmw_namespace_validation_result_string` so calling just `RCL_SET_ERROR_MSG` should be enough

Reply:
yeah, I kept it to output the value of `validation_result`, and now I tweak it to use `RCL_SET_ERROR_MSG_WITH_FORMAT_STRING` to achieve it